plugins/inputs/hystrix_stream/stream_parser.go

```diff
@@ -0,0 +1,183 @@
+	PropertyValueExecutionIsolationStrategy                       string      `json:"propertyValue_executionIsolationStrategy"`
+	PropertyValueExecutionIsolationThreadTimeoutInMilliseconds    int         `json:"propertyValue_executionIsolationThreadTimeoutInMilliseconds"`
+	PropertyValueExecutionTimeoutInMilliseconds                   int         `json:"propertyValue_executionTimeoutInMilliseconds"`
+	PropertyValueExecutionIsolationThreadInterruptOnTimeout       bool        `json:"propertyValue_executionIsolationThreadInterruptOnTimeout"`
+	PropertyValueExecutionIsolationThreadPoolKeyOverride          interface{} `json:"propertyValue_executionIsolationThreadPoolKeyOverride"`
+	PropertyValueExecutionIsolationSemaphoreMaxConcurrentRequests int         `json:"propertyValue_executionIsolationSemaphoreMaxConcurrentRequests"`
+	PropertyValueFallbackIsolationSemaphoreMaxConcurrentRequests  int         `json:"propertyValue_fallbackIsolationSemaphoreMaxConcurrentRequests"`
+	PropertyValueMetricsRollingStatisticalWindowInMilliseconds    int         `json:"propertyValue_metricsRollingStatisticalWindowInMilliseconds"`
+	PropertyValueRequestCacheEnabled                              bool        `json:"propertyValue_requestCacheEnabled"`
+	PropertyValueRequestLogEnabled                                bool        `json:"propertyValue_requestLogEnabled"`
+	ReportingHosts                                                int         `json:"reportingHosts"`
+	ThreadPool                                                    string      `json:"threadPool"`
+}
+
+var (
+	healthy       = false
+	scanner       *bufio.Scanner
+	cachedEntries []HystrixStreamEntry
+	reader        io.ReadCloser
+	cacheLock     sync.Mutex
+)
+
+func latestEntries(url string) ([]HystrixStreamEntry, error) {
+
+	if !healthy {
+		resp, err := http.Get(url)
+		if err != nil {
+			return make([]HystrixStreamEntry, 0), err
+		}
+		scanner = bufio.NewScanner(resp.Body)
+		reader = resp.Body
+		cachedEntries = make([]HystrixStreamEntry, 0)
+		go fillCacheForever(scanner)
+		healthy = true
+		log.Printf("I! Initialized hystrix-input with url : [%s]", url)
+	}
+
+	if scanner.Err() != nil {
+		log.Printf("E! Error scanning hystrix-servlet: [%v]", scanner.Err())
+		reader.Close()
+		healthy = false
+		return make([]HystrixStreamEntry, 0), scanner.Err()
+	}
+
+	defer clearCache()
+	return cachedEntries, nil
+}
+
+func clearCache() {
+	cacheLock.Lock()
+	cachedEntries = cachedEntries[:0]
+	cacheLock.Unlock()
+}
+
+func fillCacheForever(scanner *bufio.Scanner) {
+	fillCacheForeverMax(scanner, 100000)
+}
+
+func fillCacheForeverMax(scanner *bufio.Scanner, maxEntries int) {
+	newEntryCounter := 0
+	for scanner.Err() == nil {
+		chunks := streamToStrings(scanner)
+		for _, chunk := range chunks {
+			entries, err := parseChunk(chunk)
+			if err == nil {
+				for _, entry := range entries {
+					cacheLock.Lock()
```

Comment:
I don't understand the purpose of the cache, can you explain it to me?  It seems to me that it would be better to parse without the goroutine and just return the entries.

Reply:
You will probably need to use the `telegraf.ServiceInput` interface which adds Start and Stop functions.  In the Start function you can launch your goroutine that connects over http to start streaming, and also have it handle reconnects.  In the Stop function you can cancel the goroutine and wait for it to complete.  I suggest using a context.Context for this.

I recommend hanging onto the Accumulator passed into Start and calling AddFields immediately instead of having the cache.

I would say this is most like one of the queue consumer plugins such as `kafka_consumer`.